src/core_ocean/shared/mpas_ocn_vmix_gotm.F

```diff
@@ -0,0 +1,520 @@
+
+      ! set pointer for kinetic energy
+      call mpas_pool_get_array(diagnosticsPool, 'kineticEnergyCell', kineticEnergyCell)
+
+      !
+      ! set pointers for GOTM viscosity/diffusivity, TKE, rate of dissipation and turbulent length scale
+      ! (current and new state)
+      !
+      call mpas_pool_get_array(statePool, 'gotmVertViscTopOfCell', gotmVertViscTopOfCellCur, 1)
+      call mpas_pool_get_array(statePool, 'gotmVertViscTopOfCell', gotmVertViscTopOfCellNew, 2)
+      call mpas_pool_get_array(statePool, 'gotmVertDiffTopOfCell', gotmVertDiffTopOfCellCur, 1)
+      call mpas_pool_get_array(statePool, 'gotmVertDiffTopOfCell', gotmVertDiffTopOfCellNew, 2)
+      call mpas_pool_get_array(statePool, 'gotmTKETopOfCell', gotmTKETopOfCellCur, 1)
+      call mpas_pool_get_array(statePool, 'gotmTKETopOfCell', gotmTKETopOfCellNew, 2)
+      call mpas_pool_get_array(statePool, 'gotmDissTopOfCell', gotmDissTopOfCellCur, 1)
+      call mpas_pool_get_array(statePool, 'gotmDissTopOfCell', gotmDissTopOfCellNew, 2)
+      call mpas_pool_get_array(statePool, 'gotmLengthTopOfCell', gotmLengthTopOfCellCur, 1)
+      call mpas_pool_get_array(statePool, 'gotmLengthTopOfCell', gotmLengthTopOfCellNew, 2)
+      call mpas_pool_get_array(statePool, 'gotmKbTopOfCell', gotmKbTopOfCellCur, 1)
+      call mpas_pool_get_array(statePool, 'gotmKbTopOfCell', gotmKbTopOfCellNew, 2)
+      call mpas_pool_get_array(statePool, 'gotmEpsbTopOfCell', gotmEpsbTopOfCellCur, 1)
+      call mpas_pool_get_array(statePool, 'gotmEpsbTopOfCell', gotmEpsbTopOfCellNew, 2)
+
+      nCells = nCellsArray( size(nCellsArray) )
+
+      !$omp do schedule(runtime)
+      do iCell = 1, nCells
+         vertViscTopOfCell(:, iCell) = 0.0_RKIND
+         vertDiffTopOfCell(:, iCell) = 0.0_RKIND
+      end do
+      !$omp end do
+
+      ! initialize GOTM arrays at the first step
+      ! MPAS-O (1:nlev+1) <-> GOTM (nlev:0)
+      if (first) then
+         !$omp do schedule(runtime)
+         do iCell = 1, nCells
+            do k = 1, nVertLevelsP1
+               ki = nVertLevelsP1 - k
+               gotmVertViscTopOfCellCur(k, iCell) = gotm_vertVisc(ki)
+               gotmVertDiffTopOfCellCur(k, iCell) = gotm_vertDiffT(ki)
+               gotmTKETopOfCellCur(k, iCell) = gotm_TKE(ki)
+               gotmDissTopOfCellCur(k, iCell) = gotm_eps(ki)
+               gotmLengthTopOfCellCur(k, iCell) = gotm_L(ki)
+               gotmKbTopOfCellCur(k, iCell) = gotm_kb(ki)
+               gotmEpsbTopOfCellCur(k, iCell) = gotm_epsb(ki)
+               gotmVertViscTopOfCellNew(k, iCell) = gotm_vertVisc(ki)
+               gotmVertDiffTopOfCellNew(k, iCell) = gotm_vertDiffT(ki)
+               gotmDissTopOfCellNew(k, iCell) = gotm_eps(ki)
+               gotmLengthTopOfCellNew(k, iCell) = gotm_L(ki)
+               gotmKbTopOfCellNew(k, iCell) = gotm_kb(ki)
+               gotmEpsbTopOfCellNew(k, iCell) = gotm_epsb(ki)
+            end do
+         end do
+         !$omp end do
```

Comment:
At the end of a cell, edge, or vertex loop, change all omp lines to:
```
      !$omp end do
      !$omp end parallel
```

Reply:
Done